src/rustdoc-json-types/lib.rs

```diff
@@ -281,19 +281,28 @@ pub enum StructType {
     Unit,
 }
 
+#[non_exhaustive]
```

Comment:
Other things may be added as modifiers in the language, or `default` on methods could be interpreted as a modifier. This way things can be added without it being breaking

Reply:
One detail with a fallback option would be that the order is not quite arbitrary. The grammar apppears to be:

> AsyncConstQualifiers? unsafe? (extern Abi?)?

So while technically permitting the forward compatible addition of new qualifiers, a pretty-printed might still be confused by their order and not produce syntactically valid method/type declarations.

Reply:
True. Joshua said that we should use sets over vecs specifically so order is not guaranteed as part of the contract. I think that plus `non_exhaustive` means it's at least fairly obvious that we're not guaranteeing anything (all matches will require a `_` part), the only issue is if people think we're implying something about the syntax.